// gitclaw/dist/tool-factory.js
import { buildTypeboxSchema } from "./tool-loader.js";
const TOOL_DEFAULTS = {
    isConcurrencySafe: false,
    isReadOnly: false,
    isDestructive: false,
    maxResultSizeChars: 50000,
};
/**
 * Build a tool with fail-closed defaults and result truncation.
 * Mirrors Claude Code's buildTool() pattern.
 */
export function buildTool(def) {
    const metadata = { ...TOOL_DEFAULTS, ...def.metadata };
    const schema = buildTypeboxSchema(def.parameters);
    return {
        name: def.name,
        label: def.name,
        description: def.description,
        parameters: schema,
        metadata,
        async execute(_toolCallId, params, signal) {
            let result = await def.execute(params, signal);
            if (result.length > metadata.maxResultSizeChars) {
                result = result.slice(0, metadata.maxResultSizeChars) +
                    `\n\n[Truncated: ${result.length} chars total, showing first ${metadata.maxResultSizeChars}]`;
            }
            return { content: [{ type: "text", text: result }], details: undefined };
        },
    };
}
/**
 * Get metadata for a tool, returning fail-closed defaults if not set.
 */
export function getToolMetadata(tool) {
    return tool.metadata ?? { ...TOOL_DEFAULTS };
}
